### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by gedewon**
## on this PR I   implement 

- [x] Remove all hardcoded items from the tasks array.
- [x] Create a new JavaScript file for the new functionality.
- [x] Implement a function for editing task descriptions.
- [x] Implement a function for adding a new task (add a new element to the array).
- [x] By default new tasks should have the property completed set to false and the property index set to the value of the - [x] new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- [x] Implement a function for deleting a task (remove an element from the array).
- [x] Deleting a task should update all remaining items' indexes, so they represent the current list order and are - [x] unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
- [x] All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,20 +1,36 @@
...
+    });
+  });
+}
+
+window.onload = init();
+
+taskInput.addEventListener('change', (e) => {
+  todoList.addTask(e.target.value, todoList.getIndex());
+  init();
+});
```
Comment (ahmetbozaci) [inline src/index.js]: - [ ] I can add todo with a click anywhere. Even ı can add with a click on the clear button. Please fix it.

https://user-images.githubusercontent.com/38635158/165648300-b6f7ba1d-6ff6-4950-be4e-cfa3cb7d3cbe.mp4

File: src/index.js [hunk 2]
```diff
@@ -1,20 +1,36 @@
...
-window.onload = () => {
-  const root = document.querySelector('.todos-list');
+function init() {
   root.innerHTML = render();
-};
+  document.querySelectorAll('.move-delete').forEach((el) => {
+    el.addEventListener('click', (e) => {
+      todoList.remove(e.target.id);
+    });
+  });
```
Comment (ahmetbozaci) [inline src/index.js:25]: You are using the three-dot icon to delete the todo.  The clear button is not working.

- [ ] After selecting the todo, ıt should be marked with a line and ı should delete it with a click on the clear button. Please fix it. Check the [video](https://www.youtube.com/watch?v=AcUd-_Yjjqg).

image

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @Gedewon  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
- Your github-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._


___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- Your commit messages should always start with capital letters and they should be an imperative tense. I see some of them are not like that. Instead of writing `Added style or Created the file` you should write `Add style or Create the file`
- ***I write it as optional. But this is a [general requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md#:~:text=The%20commit%20message%20should%20be%20written%20in%20the%20imperative%20tense%20and%20capitalized) for every project. So you should pay attention to this in your next projects.***

**OPTIONAL**

- You can change them by using `git-rebase`. If you want to know how to change ı leave a few resources.
- [Video - How to change commit messages](https://www.loom.com/share/00771f369b7b47fc968c019ecf0d3ba6)
- [Article - How to create good commit messages](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8/)
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: src/index.js (hunk 1 above)
Comment (gedewon) [reply src/index.js]: Thank you for your feedback!!

**Review by zainsadaqat — APPROVED** [review]
# Hello @Gedewon  😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

## Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**